### DISCUSSION TITLE
Callback query

### DISCUSSION THREAD

**Discussion by senconscious** [visciang/telegram#134] [Q&A]
How can i handle callback query with ChatBot ? Tried to inspect it in handle_update function, but  there was nothing except info log in console, that: "Dropped update without chat".

**Comment by senconscious** [comment]
Seems like module Telegram.Bot.ChatBot.Chat.Session.Server when trying to extract chat seeks chat field, but from callback query the chat is under chat_instance key

**Comment by senconscious** [comment]
Not exactly, but still. Problem is that chat field when update type is callback_query is under message key. Need to fix get_chat/1 from Telegram.Bot.Utils module to work properly with callback_query case. F.E: 

```
def get_chat(update) do
    Enum.find_value(update, fn
      {_update_type, %{"chat" => %{"id" => _} = chat}} ->
        {:ok, chat}

      {"callback_query", %{"message" => %{"chat" => %{"id" => _} = chat}}} ->
        {:ok, chat}

      _ ->
        nil
    end)
  end
```

**Reply by visciang** [reply]
Fixed in #136
